pkgs/development/web/nodejs/darwin-sandbox.patch

```diff
@@ -0,0 +1,207 @@
+From 43509c3d7d805baa2fb393b0420297d34fe398de Mon Sep 17 00:00:00 2001
+From: Ivan Trubach <mr.trubach@icloud.com>
+Date: Fri, 20 Oct 2023 15:57:16 +0300
+Subject: [PATCH] fix: do not assume that /usr/bin/env exists on macOS
+
+For example, when building in Nix sandbox, /usr/bin/env is not
+available. This change also makes gyp tool execution consistent with
+gyp-win-tool that uses sys.executable on Windows.
+---
+ pylib/gyp/generator/make.py  | 19 ++++++++++---------
+ pylib/gyp/generator/ninja.py | 29 ++++++++++++++++-------------
+ pylib/gyp/xcode_emulation.py |  9 ++++++---
+ 3 files changed, 32 insertions(+), 25 deletions(-)
+
+diff --git a/pylib/gyp/generator/make.py b/pylib/gyp/generator/make.py
+index 1b997494..a964294c 100644
+--- a/pylib/gyp/generator/make.py
++++ b/pylib/gyp/generator/make.py
+@@ -25,6 +25,7 @@
+ import os
+ import re
+ import subprocess
++import sys
+ import gyp
+ import gyp.common
+ import gyp.xcode_emulation
+@@ -207,7 +208,7 @@ def CalculateGeneratorInputInfo(params):
+ 
+ LINK_COMMANDS_MAC = """\
+ quiet_cmd_alink = LIBTOOL-STATIC $@
+-cmd_alink = rm -f $@ && ./gyp-mac-tool filter-libtool libtool $(GYP_LIBTOOLFLAGS) -static -o $@ $(filter %.o,$^)
++cmd_alink = rm -f $@ && %(python)s gyp-mac-tool filter-libtool libtool $(GYP_LIBTOOLFLAGS) -static -o $@ $(filter %%.o,$^)
+ 
+ quiet_cmd_link = LINK($(TOOLSET)) $@
+ cmd_link = $(LINK.$(TOOLSET)) $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -o "$@" $(LD_INPUTS) $(LIBS)
+@@ -217,7 +218,7 @@ def CalculateGeneratorInputInfo(params):
+ 
+ quiet_cmd_solink_module = SOLINK_MODULE($(TOOLSET)) $@
+ cmd_solink_module = $(LINK.$(TOOLSET)) -bundle $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -o $@ $(filter-out FORCE_DO_CMD, $^) $(LIBS)
+-"""  # noqa: E501
++""" % {'python': sys.executable}  # noqa: E501
+ 
+ LINK_COMMANDS_ANDROID = """\
+ quiet_cmd_alink = AR($(TOOLSET)) $@
+@@ -600,14 +601,14 @@ def CalculateGeneratorInputInfo(params):
+ # Use $(4) for the command, since $(2) and $(3) are used as flag by do_cmd
+ # already.
+ quiet_cmd_mac_tool = MACTOOL $(4) $<
+-cmd_mac_tool = ./gyp-mac-tool $(4) $< "$@"
++cmd_mac_tool = %(python)s gyp-mac-tool $(4) $< "$@"
```

Comment:
Looks like doing `patchShebangs` could also work?

Reply:
Yes, but then Python would be in Node.js closure for no reason.
Relevant file in the output: `lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/mac_tool.py`.

We can remove references or replace with original shebang, but IMO that would be a workaround while the patch fixes the problem at its core.